src/pkg/services/m365/api/lists.go

```diff
@@ -218,44 +332,138 @@ func ToListable(orig models.Listable, displayName string) models.Listable {
 func cloneColumnDefinitionable(orig models.ColumnDefinitionable) models.ColumnDefinitionable {
 	newColumn := models.NewColumnDefinition()
 
+	// column attributes
+	newColumn.SetName(orig.GetName())
+	newColumn.SetOdataType(orig.GetOdataType())
+	newColumn.SetPropagateChanges(orig.GetPropagateChanges())
+	newColumn.SetReadOnly(orig.GetReadOnly())
+	newColumn.SetRequired(orig.GetRequired())
 	newColumn.SetAdditionalData(orig.GetAdditionalData())
-	newColumn.SetBoolean(orig.GetBoolean())
-	newColumn.SetCalculated(orig.GetCalculated())
-	newColumn.SetChoice(orig.GetChoice())
-	newColumn.SetColumnGroup(orig.GetColumnGroup())
-	newColumn.SetContentApprovalStatus(orig.GetContentApprovalStatus())
-	newColumn.SetCurrency(orig.GetCurrency())
-	newColumn.SetDateTime(orig.GetDateTime())
-	newColumn.SetDefaultValue(orig.GetDefaultValue())
 	newColumn.SetDescription(orig.GetDescription())
 	newColumn.SetDisplayName(orig.GetDisplayName())
-	newColumn.SetEnforceUniqueValues(orig.GetEnforceUniqueValues())
-	newColumn.SetGeolocation(orig.GetGeolocation())
+	newColumn.SetSourceColumn(orig.GetSourceColumn())
+	newColumn.SetSourceContentType(orig.GetSourceContentType())
 	newColumn.SetHidden(orig.GetHidden())
-	newColumn.SetHyperlinkOrPicture(orig.GetHyperlinkOrPicture())
 	newColumn.SetIndexed(orig.GetIndexed())
 	newColumn.SetIsDeletable(orig.GetIsDeletable())
 	newColumn.SetIsReorderable(orig.GetIsReorderable())
 	newColumn.SetIsSealed(orig.GetIsSealed())
-	newColumn.SetLookup(orig.GetLookup())
-	newColumn.SetName(orig.GetName())
-	newColumn.SetNumber(orig.GetNumber())
-	newColumn.SetOdataType(orig.GetOdataType())
-	newColumn.SetPersonOrGroup(orig.GetPersonOrGroup())
-	newColumn.SetPropagateChanges(orig.GetPropagateChanges())
-	newColumn.SetReadOnly(orig.GetReadOnly())
-	newColumn.SetRequired(orig.GetRequired())
-	newColumn.SetSourceColumn(orig.GetSourceColumn())
-	newColumn.SetSourceContentType(orig.GetSourceContentType())
-	newColumn.SetTerm(orig.GetTerm())
-	newColumn.SetText(orig.GetText())
-	newColumn.SetThumbnail(orig.GetThumbnail())
 	newColumn.SetTypeEscaped(orig.GetTypeEscaped())
-	newColumn.SetValidation(orig.GetValidation())
+	newColumn.SetColumnGroup(orig.GetColumnGroup())
+	newColumn.SetEnforceUniqueValues(orig.GetEnforceUniqueValues())
+
+	// column types
+	setColumnType(newColumn, orig)
+
+	// Requires nil checks to avoid Graph error: 'General exception while processing'
+	defaultValue := orig.GetDefaultValue()
+	if defaultValue != nil {
+		newColumn.SetDefaultValue(defaultValue)
+	}
+
+	validation := orig.GetValidation()
+	if validation != nil {
+		newColumn.SetValidation(validation)
+	}
 
 	return newColumn
 }
 
+func setColumnType(newColumn *models.ColumnDefinition, orig models.ColumnDefinitionable) {
```

Comment:
I think this func could be simplified a lot by using a blank switch statement.  ex:
```go
switch {
case orig.GetText() != nil:
  newColumn.SetText(orig.GetText())
case orig.GetBoolean() != nil:
  newColumn.SetText(orig.GetBoolean())
...
default:
  newColumn.SetText(models.NewTextColumn())
}
```